src/utils/imageOutputCuller.ts

```diff
@@ -0,0 +1,59 @@
+const LineSplitter = /\r\n?|\n/;
+const LineParser = /^[-\w]+-sha256:\w+:\s*(\w+)\s*\|.*?\|/;
+const SummaryLine = /^elapsed:.*total:/;
+
+export default class ImageOutputCuller {
+  private buffering: boolean;
+  readonly lines: string[];
+  private summaryLine: string;
+  constructor() {
+    this.buffering = true;
+    this.lines = [];
+    this.summaryLine = '';
+  }
+
+  addData(data: string): void {
+    // TODO (possibly): Deal with partial final lines - I haven't seen this happen yet
+    const lines = data.split(LineSplitter);
+
+    lines.forEach((line) => {
+      if (!this.buffering) {
+        this.lines.push(line);
+      } else if (SummaryLine.test(line)) {
+        this.summaryLine = line;
+      } else if (/^\s*$/.test(line)) {
+        // do nothing
+      } else {
+        const m = LineParser.exec(line);
+
+        if (m) {
+          const idx = this.lines.findIndex((elt: string) => elt.includes(m[1]));
```

Comment:
The type (`elt: string`) here is redundant: TypeScript knows that `this.lines` is a `string[]`, and therefore `T[].findIndex(T => bool)` takes a string.

Reply:
ok